mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -156,92 +159,6 @@ impl ConnectionSettings {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Deserialize)]
-#[serde(untagged)]
-pub enum Credentials {
-    Anonymous(String),
-    PlainText(AuthenticationSettings),
-    Provider(CredentialProviderSettings),
-}
-
-#[derive(Debug, Clone, Default, PartialEq, Deserialize)]
-pub struct AuthenticationSettings {
-    client_id: String,
-
-    username: String,
-
-    password: String,
-}
-
-impl AuthenticationSettings {
-    pub fn new(client_id: String, username: String, password: String) -> Self {
-        Self {
-            client_id,
-            username,
-            password,
-        }
-    }
-
-    pub fn client_id(&self) -> &str {
-        &self.client_id
-    }
-
-    pub fn username(&self) -> &str {
-        &self.username
-    }
-
-    pub fn password(&self) -> &str {
-        &self.password
-    }
-}
-
-#[derive(Debug, Clone, PartialEq, Deserialize)]
-pub struct CredentialProviderSettings {
-    #[serde(rename = "iotedge_iothubhostname")]
-    iothub_hostname: String,
-
-    #[serde(rename = "iotedge_gatewayhostname")]
-    gateway_hostname: String,
-
-    #[serde(rename = "iotedge_deviceid")]
-    device_id: String,
-
-    #[serde(rename = "iotedge_moduleid")]
-    module_id: String,
-
-    #[serde(rename = "iotedge_modulegenerationid")]
-    generation_id: String,
-
-    #[serde(rename = "iotedge_workloaduri")]
-    workload_uri: String,
-}
-
-impl CredentialProviderSettings {
-    pub fn iothub_hostname(&self) -> &str {
-        &self.iothub_hostname
-    }
-
-    pub fn gateway_hostname(&self) -> &str {
-        &self.gateway_hostname
-    }
-
-    pub fn device_id(&self) -> &str {
-        &self.device_id
-    }
-
-    pub fn module_id(&self) -> &str {
-        &self.module_id
-    }
-
-    pub fn generation_id(&self) -> &str {
-        &self.generation_id
-    }
-
-    pub fn workload_uri(&self) -> &str {
-        &self.workload_uri
-    }
-}
-
```

Comment:
I thought it was a good idea to move these settings into the util crate since test modules will be using these too.